pkg/rvasp/rvasp.go

```diff
@@ -226,16 +226,27 @@ func (s *Server) Transfer(ctx context.Context, req *pb.TransferRequest) (reply *
 		return nil, status.Errorf(codes.FailedPrecondition, "unknown originator policy '%s' for wallet '%s'", policy, account.WalletAddress)
 	}
 
-	if transferError != nil {
-		log.Debug().Err(transferError).Msg("transfer failed")
-		xfer.SetState(pb.TransactionState_FAILED)
-	}
-
-	// Return the transfer response
+	// Build the transfer response
 	reply = &pb.TransferReply{
 		Transaction: xfer.Proto(),
 	}
 
+	// Handle rVASP errors and TRISA protocol errors
+	if transferError != nil {
+		switch err := transferError.(type) {
+		case *protocol.Error:
+			log.Warn().Str("message", err.Error()).Msg("TRISA protocol error while performing transfer")
+			reply.Error = &pb.Error{
+				Message: err.Error(),
+			}
```

Comment:
Could we also include the code here as well?

```suggestion
			reply.Error = &pb.Error{
				Code: err.Code,			
				Message: err.Message,
			}
```